### ISSUE TITLE
Cannot join Saved Questions that themselves contains joins

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#12928]
**Describe the bug**
When trying to join two Saved Questions, which themselves are using joins, then the query fails.

**To Reproduce**
1. Create question Q1 - Orders join with Products and People, summarized by Count, grouped by Category and Source
![image](https://user-images.githubusercontent.com/1447303/87708713-6e290480-c7a3-11ea-861a-72d09dac8c30.png)
2. Create question Q2 - Reviews join with Products, summarized by Average Rating, grouped by Category
![image](https://user-images.githubusercontent.com/1447303/87708937-c8c26080-c7a3-11ea-983e-e53c8d868d3f.png)
3. Custom question > Saved Questions > Q1 and join Q2 on Category
![image](https://user-images.githubusercontent.com/1447303/87709048-f5767800-c7a3-11ea-853c-b7cb96f524e9.png)
4. Visualize errors with: `Bad :joined-field clause: join with alias 'Product' does not exist. Found: #{"Question 241"}` (full trace below)
5. Changing Q2 to not have a join like Products summarized by Average Rating grouped by Category, then it's possible to get results in step 4
![image](https://user-images.githubusercontent.com/1447303/87709318-5ef68680-c7a4-11ea-9572-b0ecb30371af.png)

<details><summary>stacktrace</summary>

```
07-16 20:25:02 INFO api.dataset :: Source query for this query is Card 240
07-16 20:25:02 INFO middleware.fetch-source-query :: Fetched source query from Card 240: 
 {:source-table 11,
 :aggregation [[:count]],
 :joins
 [{:fields :all, :source-table 10, :condition [:= [:field-id 82] [:joined-field "Product" [:field-id 105]]], :alias "Product"}
  {:fields :all, :source-table 12, :condition [:= [:field-id 80] [:joined-field "User" [:field-id 92]]], :alias "User"}],
 :breakout [[:joined-field "Product" [:field-id 107]] [:joined-field "User" [:field-id 95]]]}

07-16 20:25:02 INFO middleware.fetch-source-query :: Fetched source query from Card 241: 
 {:source-table 13,
 :joins [{:fields :all, :source-table 10, :condition [:= [:field-id 110] [:joined-field "Product" [:field-id 105]]], :alias "Product"}],
 :aggregation [[:avg [:field-id 115]]],
 :breakout [[:joined-field "Product" [:field-id 107]]]}

07-16 20:25:02 INFO middleware.fetch-source-query :: Fetched source query from Card 240: 
 {:source-table 11,
 :aggregation [[:count]],
 :joins
 [{:fields :all, :source-table 10, :condition [:= [:field-id 82] [:joined-field "Product" [:field-id 105]]], :alias "Product"}
  {:fields :all, :source-table 12, :condition [:= [:field-id 80] [:joined-field "User" [:field-id 92]]], :alias "User"}],
 :breakout [[:joined-field "Product" [:field-id 107]] [:joined-field "User" [:field-id 95]]]}

07-16 20:25:02 INFO middleware.fetch-source-query :: Fetched source query from Card 241: 
 {:source-table 13,
 :joins [{:fields :all, :source-table 10, :condition [:= [:field-id 110] [:joined-field "Product" [:field-id 105]]], :alias "Product"}],
 :aggregation [[:avg [:field-id 115]]],
 :breakout [[:joined-field "Product" [:field-id 107]]]}

07-16 20:25:02 ERROR middleware.catch-exceptions :: Error processing query: null
{:database_id 4,
 :started_at (t/zoned-date-time "2020-07-16T20:25:02.021786+02:00[Europe/Copenhagen]"),
 :json_query
 {:type "query",
  :query
  {:source-table "card__240",
   :joins
   [{:fields "all",
     :source-table "card__241",
     :condition ["=" ["field-literal" "CATEGORY" "type/Text"] ["joined-field" "Question 241" ["field-literal" "CATEGORY" "type/Text"]]],
     :alias "Question 241",
     :strategy "left-join"}],
   :limit 10},
  :database 4,
  :parameters [],
  :middleware {:add-default-userland-constraints? true}},
 :native nil,
 :status :failed,
 :class java.lang.IllegalArgumentException,
 :stacktrace
 ["--> query_processor.middleware.resolve_joins$check_join_aliases.invokeStatic(resolve_joins.clj:139)"
  "query_processor.middleware.resolve_joins$check_join_aliases.invoke(resolve_joins.clj:133)"
  "query_processor.middleware.resolve_joins$fn__44312$resolve_joins_in_mbql_query__44317$fn__44321.invoke(resolve_joins.clj:148)"
  "query_processor.middleware.resolve_joins$fn__44312$resolve_joins_in_mbql_query__44317.invoke(resolve_joins.clj:142)"
  "query_processor.middleware.resolve_joins$resolve_joins_in_mbql_query_all_levels.invokeStatic(resolve_joins.clj:165)"
  "query_processor.middleware.resolve_joins$resolve_joins_in_mbql_query_all_levels.invoke(resolve_joins.clj:163)"
  "query_processor.middleware.resolve_joins$resolve_joins_STAR_.invokeStatic(resolve_joins.clj:177)"
  "query_processor.middleware.resolve_joins$resolve_joins_STAR_.invoke(resolve_joins.clj:175)"
  "query_processor.middleware.resolve_joins$resolve_joins$fn__44345.invoke(resolve_joins.clj:183)"
  "query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__39271.invoke(add_implicit_joins.clj:245)"
  "query_processor.middleware.limit$limit$fn__43135.invoke(limit.clj:38)"
  "query_processor.middleware.format_rows$format_rows$fn__43116.invoke(format_rows.clj:81)"
  "query_processor.middleware.desugar$desugar$fn__42602.invoke(desugar.clj:22)"
  "query_processor.middleware.binning$update_binning_strategy$fn__41628.invoke(binning.clj:229)"
  "query_processor.middleware.resolve_fields$resolve_fields$fn__42144.invoke(resolve_fields.clj:24)"
  "query_processor.middleware.add_dimension_projections$add_remapping$fn__38807.invoke(add_dimension_projections.clj:272)"
  "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__39027.invoke(add_implicit_clauses.clj:147)"
  "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__39420.invoke(add_source_metadata.clj:105)"
  "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__44010.invoke(reconcile_breakout_and_order_by_bucketing.clj:98)"
  "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__41269.invoke(auto_bucket_datetimes.clj:125)"
  "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__42191.invoke(resolve_source_table.clj:46)"
  "query_processor.middleware.parameters$substitute_parameters$fn__43795.invoke(parameters.clj:97)"
  "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__42243.invoke(resolve_referenced.clj:80)"
  "query_processor.middleware.expand_macros$expand_macros$fn__42858.invoke(expand_macros.clj:158)"
  "query_processor.middleware.add_timezone_info$add_timezone_info$fn__39451.invoke(add_timezone_info.clj:15)"
  "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__47166.invoke(splice_params_in_response.clj:32)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__44021$fn__44025.invoke(resolve_database_and_driver.clj:33)"
  "driver$do_with_driver.invokeStatic(driver.clj:61)"
  "driver$do_with_driver.invoke(driver.clj:57)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__44021.invoke(resolve_database_and_driver.clj:27)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__43064.invoke(fetch_source_query.clj:243)"
  "query_processor.middleware.store$initialize_store$fn__47175$fn__47176.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:46)"
  "query_processor.store$do_with_store.invoke(store.clj:40)"
  "query_processor.middleware.store$initialize_store$fn__47175.invoke(store.clj:10)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__42120.invoke(cache.clj:209)"
  "query_processor.middleware.validate$validate_query$fn__47184.invoke(validate.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__43162.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__39289.invoke(add_rows_truncated.clj:36)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__47151.invoke(results_metadata.clj:128)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__42479.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__43884.invoke(process_userland_query.clj:136)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__42422.invoke(catch_exceptions.clj:174)"
  "query_processor.reducible$async_qp$qp_STAR___38090$thunk__38091.invoke(reducible.clj:101)"
  "query_processor.reducible$async_qp$qp_STAR___38090.invoke(reducible.clj:107)"
  "query_processor.reducible$sync_qp$qp_STAR___38099$fn__38102.invoke(reducible.clj:133)"
  "query_processor.reducible$sync_qp$qp_STAR___38099.invoke(reducible.clj:132)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:218)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:214)"
  "query_processor$fn__47350$process_query_and_save_execution_BANG___47359$fn__47362.invoke(query_processor.clj:233)"
  "query_processor$fn__47350$process_query_and_save_execution_BANG___47359.invoke(query_processor.clj:225)"
  "query_processor$fn__47394$process_query_and_save_with_max_results_constraints_BANG___47403$fn__47406.invoke(query_processor.clj:245)"
  "query_processor$fn__47394$process_query_and_save_with_max_results_constraints_BANG___47403.invoke(query_processor.clj:238)"
  "api.dataset$fn__50692$fn__50695.invoke(dataset.clj:52)"
  "query_processor.streaming$streaming_response_STAR_$fn__34128$fn__34129.invoke(streaming.clj:73)"
  "query_processor.streaming$streaming_response_STAR_$fn__34128.invoke(streaming.clj:72)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:66)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:64)"
  "async.streaming_response$do_f_async$fn__23118.invoke(streaming_response.clj:85)"],
 :context :ad-hoc,
 :error "Bad :joined-field clause: join with alias 'Product' does not exist. Found: #{\"Question 241\"}",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :data {:rows [], :cols []}}

07-16 20:25:02 DEBUG middleware.log :: POST /api/dataset 202 [ASYNC: completed] 254.1 ms (13 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (136 total active threads) Queries in flight: 0 (0 queued)
```

</details>

**Information about your Metabase Installation:**
Metabase 1.35.4.1 and `master`

**Additional context**
Should these Saved Questions not be shown at all? Since that seems like how it works for SQL-based questions #11194
If implicit joins (linked in Data Model) are used, then the error is seems similar to #12762

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by Sidhantp12** [comment]
![image](https://user-images.githubusercontent.com/54805671/89873002-128f4280-dbba-11ea-9fdd-3dfea3d9246c.png)

Trying to make a custom question out of these two (Previously saved questions which has joins in it)

![image](https://user-images.githubusercontent.com/54805671/89873103-35b9f200-dbba-11ea-837d-d9c6ef7fc468.png)

Following error I got. I guess, its related to this bug !!

**Comment by christiankf** [comment]
I run into the same issue. One workaround that seems to work is to convert the query to SQL and then use this question as a base. Without further modification that got rid of the error.

**Comment by soltanianalytics** [comment]
@christiankf thanks, that workaround worked for me!

Note: In [this issue](https://github.com/metabase/metabase/issues/13737) I have a case where the first table is a table in the database and the second question is a saved question _without_ at join. It still fails, unless I convert the second question into an SQL based question.

**Comment by camsaul** [comment]
Fixed by #16254
